blockpool/peers.go

```diff
@@ -218,17 +250,11 @@ func (self *peers) addPeer(
 	if found {
 		// when called on an already connected peer, it means a newBlockMsg is received
 		// peer head info is updated
-		p.lock.Lock()
-		if p.currentBlockHash != currentBlockHash {
-			previousBlockHash = p.currentBlockHash
-			plog.Debugf("addPeer: Update peer <%s> with td %v and current block %s (was %v)", id, td, hex(currentBlockHash), hex(previousBlockHash))
-			p.setChainInfo(td, currentBlockHash)
-
-			self.status.lock.Lock()
-			self.status.values.NewBlocks++
-			self.status.lock.Unlock()
-		}
-		p.lock.Unlock()
+		p.setChainInfo(td, currentBlockHash)
+		// FIXME: only count the same block once
+		self.status.lock.Lock()
```

Comment:
This is a non-blocking suggestion for this PR

You should consider wrapping these things up in accessors (or consider using `atomic.AddInt32`)